gulpfile.js

```diff
@@ -335,24 +335,35 @@ function checkChromePreferencesFile(chromePrefsPath, webPrefs) {
   const webPrefsKeys = Object.keys(webPrefs);
   webPrefsKeys.sort();
 
-  if (webPrefsKeys.length !== chromePrefsKeys.length) {
-    console.log("Warning: Pref objects doesn't have the same length.");
-    return false;
-  }
-
   let ret = true;
-  for (let i = 0, ii = webPrefsKeys.length; i < ii; i++) {
-    const value = webPrefsKeys[i];
-    if (chromePrefsKeys[i] !== value) {
+  // Verify that every entry in webPrefs is also in preferences_schema.json.
+  for (const [key, value] of Object.entries(webPrefs)) {
+    if (!chromePrefsKeys.includes(key)) {
+      // Note: this would also reject keys that are present but marked as
+      // DEPRECATED. A key should not be marked as DEPRECATED if it is still
+      // listed in webPrefs.
       ret = false;
       console.log(
-        `Warning: not the same keys: ${chromePrefsKeys[i]} !== ${value}`
+        `Warning: ${chromePrefsPath} does not contain entry for pref: ${key}`
       );
-    } else if (chromePrefs.properties[value].default !== webPrefs[value]) {
+    } else if (chromePrefs.properties[key].default !== value) {
+      ret = false;
+      console.log(
+        `Warning: not the same values (for "${key}"): ` +
+          `${chromePrefs.properties[key].default} !== ${value}`
+      );
+    }
+  }
+
+  // Verify that preferences_schema.json does not contain entries that are not
+  // in webPrefs (app_options.js).
+  for (const key of chromePrefsKeys) {
+    if (!(key in webPrefs)) {
       ret = false;
       console.log(
-        `Warning: not the same values (for "${value}"): ` +
-          `${chromePrefs.properties[value].default} !== ${webPrefs[value]}`
+        `Warning: ${chromePrefsPath} contains unrecognized pref: ${key}. ` +
+          `Remove it, or prepend "DEPRECATED: " and add migration logic to ` +
```

Comment:
Looking at the code above and also the `preferences_schema.json` file this looks like a small typo here:
```suggestion
          `Remove it, or prepend "DEPRECATED. " and add migration logic to ` +
```

Reply:
Done. While I am at it, I also added "an" 2x in the output messages where it made sense.